zebra/zebra_dplane.c

```diff
@@ -6495,6 +6500,9 @@ void zebra_dplane_ns_enable(struct zebra_ns *zns, bool enabled)
 		/* Stop reading, free memory */
 		zns_info_list_del(&zdplane_info.dg_zns_list, zi);
 
+		/* Clean up outstanding update contexts */
+		dplane_clean_ctx_queue(dplane_context_cb_ns_id, &zns->ns_id);
```

Comment:
what if there are queued contexts representing cleanup work that needs to be done?

Reply:
`kernel_terminate()` is closing the netlink socket after this call, thus no cleanup is possible. Kernel will take care of this since the netns is being deleted.

Reply:
the kernel isn't the only dataplane, or consumer of dataplane events - I would prefer not to make that assumption

Reply:
I'm not sure. How does other types of dataplane define netns? In what circumstances are pending updates of a namespace still valid after it's deletion?

Reply:
@mjstapp Moved this to kernel-specific part. Please check.